.github/workflows/auto_cherry_pick.yml

```diff
@@ -0,0 +1,42 @@
+# CI stages to execute against all branches on PR merge
+name: auto_cherry_pick_commits
+
+on:
+  push
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  branch-matrix:
+    name: Generate a branch matrix to apply cherrypicks
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - id: set-matrix
+        run: echo "::set-output name=branches::$(git branch -rl --sort=-authordate 'origin/6.*.z' --format='%(refname:lstrip=-1)' | head -n2 | jq -cnR '[inputs | select(length>0)]')"
```

Comment:
Have you tried on freshly cloned Robottelo ?

Reply:
I tried it on my existing clone of robottelo, where the results are the same.

@jyejare I wonder why you considering N-1 and N-2 cherrypicks at the same time and using this complex script to generate a matrix of branches to cherrypick, instead, you could've put hardcode N-1 version(6.10.z) here, and when N-1 cherrypick-PR will be open, where if needed you could add `Cherrypick` label again, and it should open its N-1 version(6.9.z) PR, will it work?
@JacobCallahan and @mshriver Is there anything you would like to add or correct?

Reply:
@Gauravtalreja1 I suppose it checks the `origin` remote, so possibly in your local clone that points to your fork of robottelo -- if you haven't ever pushed those z branches there, they won't be there. Maybe try with `upstream`

@jyejare I would just make sure that the `jq` tool is installed by default on ubuntu-latest

Reply:
@pondrejk Thanks for the review! Yes jq works as expected at least I tested on locally on my system and it works just fine. See Description of this PR I updated.

Reply:
@Gauravtalreja1 When I developed this GHA, I didn't have https://github.com/SatelliteQE/robottelo/pull/9762 in mind which will enforce any PR to cherrypick or no-cherrypick. We saw how problematic it is when we dont cherrypick to n-2 (currently 6.9) that people usually forget about back-porting their changes to n-2 but they do it easily for n-1. And hence n-2 for `master branch(it would be auto reduced to -1 for below branches like 6.10 -> 6.9 and 6.9 don't cherrypick)` was my default go-to option for cherrypicking.

I dont care about script complexity unless its working as expected and the script is working for me perfect, you can see the description updated.

I wanted to avoid the hardcoding for a reason that branching should also change that version when we create a new branch like 6.11.z forgetting that would cherrypick to 6.10 directly from 6.12. But since other GHA is also hardcoding those values and since now we have forced cherrypick labelling on PRs , I am OK to go for hardcoding values. I would like to hear what others are saying @JacobCallahan @mshriver @pondrejk !

Reply:
I'd go with the current line, if it breaks for some reason in the future we would fix it but that's hypothetical. With hardcoded values we have certainty that we will have to update the wf regularly, which is quite easy to forget about.